mrp_bom_nested_bom_generator/models/mrp_nested_bom.py

```diff
@@ -0,0 +1,188 @@
+from typing import Any, Dict, Set
+
+from odoo import _, api, fields, models
+
+
+class MrpNestedBomLine(models.Model):
+    _name = "mrp.nested.bom"
+    _description = "MRP Nested BOM Line"
+
+    sequence = fields.Integer(string="Sequence", default=10)
+
+    bom_id = fields.Many2one(
+        comodel_name="mrp.bom",
+        string="MRP BOM",
+        required=True,
+    )
+
+    parent_id = fields.Many2one(
+        comodel_name="product.template",
+        string="Product Template",
+        related="bom_id.product_tmpl_id",
+    )
+
+    all_attribute_ids = fields.Many2many(
+        "product.attribute",
+        "all_attribute_ids_ref",
+        string="Parent Product Template attribute",
+        compute="_compute_all_attribute_ids",
+        store=True,
+    )
+
+    product_tmpl_id = fields.Many2one(
+        comodel_name="product.template",
+        string="Product",
+        domain="[('id', '!=', parent_id)]",
+    )
+    product_qty = fields.Float(
+        string="Qty",
+        default=1,
+        required=True,
+    )
+    uom_id = fields.Many2one(
+        "uom.uom",
+        string="Unit of Measure",
+        related="product_tmpl_id.uom_id",
+        readonly=True,
+    )
+
+    attribute_ids = fields.Many2many(
+        "product.attribute",
+        "attribute_ids_ref",
+        string="Attributes",
+    )
+
+    @api.onchange("product_tmpl_id")
+    def _onchange_product_tmpl_id(self) -> None:
+        """
+        Change attribute field at changing product template field
+        :return None
+        """
+        self.attribute_ids = self.product_tmpl_id.attribute_line_ids.mapped(
+            "attribute_id"
+        )
+
+    @api.depends("bom_id.nested_bom_ids", "parent_id.attribute_line_ids")
+    def _compute_all_attribute_ids(self) -> None:
+        """
+        Compute product attributes by parent product and component
+        :return None
+        """
+        for rec in self:
+            default_attributes = rec.product_tmpl_id.attribute_line_ids.mapped(
+                "attribute_id"
+            )
+            parent_attributes = rec.parent_id.attribute_line_ids.mapped("attribute_id")
+            attributes_ids = default_attributes | parent_attributes
+            rec.all_attribute_ids = attributes_ids
+
+    def write(self, vals):
+        self.mapped("bom_id").write({"changed_nested_bom": True})
+        return super(MrpNestedBomLine, self).write(vals)
+
+    @api.model
+    def create_product(self, bom_id: int) -> int:
+        """
+        Create product by parent product
+        :param int bom_id: mrp.bom record id
+        :return new product template id
```

Comment:
This is confusing. What are you returning? id of the new product template? or new product_template record?

Reply:
Resolved